src/ui/dialogs/contactsearch.ts

```diff
@@ -0,0 +1,253 @@
+    let fields = null;
+    if ((items=$(result).find('query').children('item')).length>0)//Simple results
+    {
+
+        header.push(Translation.t('First')); //first
+        header.push(Translation.t('Last')); //last
+        header.push(Translation.t('Nick')); //nick
+        header.push(Translation.t('Email')); //email
+
+        simple=true;
+    }
+    else
+        if ($(result).find('query').children('x').length>0)//Extended results
+        {
+            items=$(result).find('query').children('x').children('item');
+            if (items.length===0)
+            {
+                //NO RESULTS FOUND!
+                dialog.close();
+                alert(Translation.t('no_results'));
+                return;
+            }
+            fields = $(result).find('query').find('reported').children('field');
+            let i=0;
+            for (;i<fields.length;i++)
+            {
+                header.push($(fields[i]).attr('label'));
+            }
+        }
+        else
+        {
+            //NO RESULTS FOUND!
+            dialog.close();
+            alert(Translation.t('no_results'));
+            return;
+        }
+
+    let html = '<table id="jsxc-searchresults" class="jsxc-searchresults-table"><thead class="jsxc-searchresults-table-head"><tr class="jsxc-searchresults-table-head-row">';
+    let i=0;
+    for (;i<header.length;i++)
+    {
+        html+='<th>'+header[i]+'</th>';
+    }
+    html+='</tr></thead><tbody class="jsxc-searchresults-table-body">';
+
+    i=0;
+    for (;i<items.length;i++)
+    {
+        html+='<tr class="jsxc-searchresults-table-body-row '+(i%2===0?'jsxc-searchresults-odd':'jsxc-searchresults-even')+'"><td><input class="jsxc-searchresults-add" type="checkbox" name="jsxc-add-contact" value="'+i+'"/></td>';
+        if (simple)
+        {
+            html+='<td>';
+            if ($(items[i]).children('first').length>0)
+            {
+                html+=$(items[i]).children('first').text();
+            }
+            html+='</td><td>';
+            if ($(items[i]).children('last').length>0)
+            {
+                html+=$(items[i]).children('last').text();
+            }
+            html+='</td><td>';
+            if ($(items[i]).children('nick').length>0)
+            {
+                html+=$(items[i]).children('nick').text();
+            }
+            html+='</td><td>';
+            if ($(items[i]).children('email').length>0)
+            {
+                html+=$(items[i]).children('email').text();
+            }
+            html+='</td>';
+        }
+        else
+        {
+            let n=0;
+            for (;n<fields.length;n++)
+            {
+                html+='<td>';
+                html+=$(items[i]).find('field[var="'+$(fields[n]).attr('var')+'"]').text();
+                html+='</td>';
+            }
+        }
+
+        html+='</tr>';
+    }
+
+    html+='</tbody></table>';
+
+    dialog.getDom().find('#content').empty().append(html);
```

Comment:
Needs a prefix and should not use an id.

Reply:
done